src/ir_passes/await.ml

```diff
@@ -370,41 +426,39 @@ and c_exp' context exp k =
     begin
       match LabelEnv.find_opt (Named id) context with
       | Some (Cont k') -> c_exp context exp1 k'
-      | Some Label -> assert false
-      | None -> assert false
+      | _ -> assert false
     end
   | PrimE (RetPrim, [exp1]) ->
     begin
       match LabelEnv.find_opt Return context with
       | Some (Cont k') -> c_exp context exp1 k'
-      | Some Label -> assert false
-      | None -> assert false
+      | _ -> assert false
     end
   | PrimE (ThrowPrim, [exp1]) ->
     begin
       match LabelEnv.find_opt Throw context with
       | Some (Cont k') -> c_exp context exp1 k'
-      | Some Label
-      | None -> assert false
+      | _ -> assert false
     end
   | AsyncE (T.Cmp, tb, exp1, typ1) ->
     assert false (* must have effect T.Triv, handled by first case *)
   | AsyncE (T.Fut, tb, exp1, typ1) ->
     (* add the implicit return label *)
     let k_ret = fresh_cont (typ exp1) T.unit in
     let k_fail = fresh_err_cont T.unit in
+    let k_clean = fresh_cont T.unit T.unit in
     let context' =
-      LabelEnv.add Return (Cont (ContVar k_ret))
-        (LabelEnv.add Throw (Cont (ContVar k_fail)) LabelEnv.empty)
+      LabelEnv.add Cleanup (Cont (ContVar k_clean))
```

Comment:
Similarly for inline `async { ... }` the send shouldn't trap and the only code executed is RTS. No user code is running. But I am adding it for symmetry reasons, without a test.

Reply:
@crusso what do you think? Could I test this somehow?